crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -422,7 +529,24 @@ impl<'a> StatementVisitor<'a> for BodyVisitor<'a> {
             _ => {}
         }
 
-        statement_visitor::walk_stmt(self, stmt);
+        visitor::walk_stmt(self, stmt);
+    }
+
+    fn visit_expr(&mut self, expr: &'a Expr) {
+        match expr {
+            Expr::Yield(ast::ExprYield {
+                range,
+                value: Some(_),
+            }) => {
+                self.yields.push(Entry { range: *range });
+            }
+            Expr::YieldFrom(ast::ExprYieldFrom { range, .. }) => {
+                self.yields.push(Entry { range: *range });
+            }
+            Expr::Lambda(_) => return,
```

Comment:
Why is `Lambda` being specifically special-cased here?

Reply:
I think the reason is because we don't want to/need to traverse into lambda bodies

Reply:
Aha, it seems that `yield`s are invalid syntax inside `lambda`s! So it makes sense that we don't need to traverse inside them. Maybe we should add a comment?

```pycon
>>> x = lambda: yield 42
  File "<stdin>", line 1
    x = lambda: yield 42
                ^^^^^
SyntaxError: invalid syntax
>>> x = lambda: yield from ()
  File "<stdin>", line 1
    x = lambda: yield from ()
                ^^^^^
SyntaxError: invalid syntax
>>> x = lambda: yield
  File "<stdin>", line 1
    x = lambda: yield
                ^^^^^
SyntaxError: invalid syntax
```

Reply:
It matches the handling of functions and classes above.

Reply:
Good point. A comment isn't necessary, then.